homeassistant/components/vegehub/config_flow.py

```diff
@@ -0,0 +1,260 @@
+"""Config flow for the VegeHub integration."""
+
+import logging
+from typing import Any
+
+from vegehub import VegeHub
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.components.webhook import (
+    async_generate_id as webhook_generate_id,
+    async_generate_url as webhook_generate_url,
+)
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import callback
+
+from .const import DOMAIN, OPTION_DATA_TYPE_CHOICES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ip_dict: dict[str, str] = {}
+
+
+class VegeHubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for VegeHub integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the VegeHub config flow."""
+        self._hub: VegeHub | None = None
+        self._hostname: str = ""
+        self._properties: dict = {}
+        self._config_url: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial confirmation step with no inputs."""
+        errors = {}
+
+        if user_input is not None:
+            if "ip_address" in user_input and self._hub is None:
+                # When the user has input the IP manually, we need to gather more information
+                # from the Hub before we can continue setup.
+                self._hub = VegeHub(str(user_input.get("ip_address")))
+
+                await self._hub.retrieve_mac_address()
```

Comment:
this sounds like it can throw?

Reply:
Yes, it's possible for that function to throw a ConnectionError. Should I be doing something differently?

Reply:
Oh, ok. I added a try/except (2369e04c6be5de82d34dc613b702e2414c1f5987)